release.sh

```diff
@@ -301,6 +302,27 @@ function build_from_source() {
   if [[ $? -ne 0 ]]; then
     abort "error building the release"
   fi
+  sign_release || abort "error signing the release"
+}
+
+# Build a release from source.
+function sign_release() {
+  ## Sign the images with cosign
+  ## For now, check if ko has created imagerefs.txt file. In the future, missing image refs will break
+  ## the release for all jobs that publish images.
+  if [[ -f "imagerefs.txt"  && -z ${SIGNING_KEY} ]]; then
+      echo "Signing Images with the key ${SIGNING_KEY}"
+      COSIGN_EXPERIMENTAL=1 cosign sign $(cat imagerefs.txt) --key gcpkms://"${SIGNING_KEY}"
```

Comment:
OK, let me do an argument here.

Of course, signing with any KMS works but I would strongly suggest moving towards keyless signing now instead of switching later. The main reason here is that you are effectively buying the project a key management problem it does not really need.

Each release will now require to have its own keys published with the artifacts (as you are doing it now by copying it to the bucket), and while this would not be that bad 10 years ago but now we have a lot of automated tooling everywhere verifying signatures which expect to find the certificate attached to the image. 

You can use the key file or the cloud kms to verify the images, but most consumers will go via the automated route and it will simply not work for them. The kyverno policy at the top of the doc you linked above is a good example: You could potentially need one such policy listing each release if you rotate the keys with regularity (which you should). Compare that with the [keyless policy in the same doc](https://kyverno.io/docs/writing-policies/verify-images/#keyless-signing-and-verification), it is a single one to verify all releases.

So tldr, keyless should be straightforward to implement and saves the project and its users some problems now but a ton in the future.

Reply:
Thanks for the comments, i'll go ahead with Keyless signing.

Infra is being created at https://github.com/knative/test-infra/pull/3516